### DISCUSSION TITLE
Installing Directus 9 thru cPanel > Setup Node.js App ?

### DISCUSSION THREAD

**Discussion by adrchen** [directus/directus#5129] [Archive]
Trying to install Directus 9 thru cPanel, under the Setup Node.js App.

I added the information, and then ran the configuration file (Run NPM Install), but nothing is happening, all I get on the URL is:

```
It works!

NodeJS 12.21.0
```

Anyone else had experience installing Directus 9 using this method? Searched and there's not much information other than someone giving up at some point.

<img width="1432" alt="Screen Shot 2021-04-17 at 9 01 15 PM" src="https://user-images.githubusercontent.com/13161663/115131815-8fcc4c00-9fc0-11eb-962f-17bcd283099d.png">

**Comment by lahdekorpi** [comment]
You can create a wrapper that runs Directus in a Passenger friendly way.

First create the `app.js` file (or modify if created by cPanel already):

```javascript
if (typeof PhusionPassenger !== 'undefined') {
  console.log('Running inside Passenger.');
} else {
  console.log('Running outside Passenger.');
}

const { default: logger } = require('directus/dist/logger');
const { default: start } = require('directus/dist/start');

start().catch(function(err) {
    logger.error("Error", { err });
});
```

Then make sure you have the packages installed (if you want a specific version, install through the command line as the "Run NPM Install" forces the latest version).

**Reply by lahdekorpi** [reply]
Lastly, add all the environment variables through the UI.
